### ISSUE TITLE
Invalid argument error in test runs

### ISSUE THREAD

**Issue by karthiknadig** [microsoft/ptvsd#1834]
## Environment data

- PTVSD version: master
- OS and version: linux
- Python version (& distribution if applicable, e.g. Anaconda): any
- Using VS Code or Visual Studio: N/A

```
Exception in thread Debuggee-187.backchannel listener:
Traceback (most recent call last):
  File "/opt/python/3.7.1/lib/python3.7/threading.py", line 917, in _bootstrap_inner
    self.run()
  File "/opt/python/3.7.1/lib/python3.7/threading.py", line 865, in run
    self._target(*self._args, **self._kwargs)
  File "/home/travis/build/microsoft/ptvsd/tests/debug/comms.py", line 44, in accept_worker
    self._socket, _ = self._server_socket.accept()
  File "/opt/python/3.7.1/lib/python3.7/socket.py", line 212, in accept
    fd, addr = self._accept()
OSError: [Errno 22] Invalid argument
```

**Comment by int19h** [comment]
Looking at the code, the only way this can happen is if `BackChannel.close()` gets invoked before (or perhaps during) `accept()` is called in `BackChannel.listen()` - i.e. if the test is being torn down almost immediately without a backchannel connection is established. 

So, `close()` should set the server socket attribute to `None` before closing it, and `listen()` should catch the error and check if socket is `None`.

**Comment by int19h** [comment]
This is tentatively fixed by 173f9b3a481abd83df8fc4c788b8b5f8c6a6dee8.

**Comment by codingpsych72** [comment]
thank you mr.int19h it was so helpful
